brave/brave-core
# Added IPNS keys settings option

Second part of https://github.com/brave/brave-browser/issues/15422
First part here https://github.com/brave/brave-core/pull/8609

- Added option to IPFS settings that links to keys management subpage, the option is visible only if local node is enabled
- Added APIs to add/remove/fetch keys from node, available only on settings page
- Added observers to switch UI when the node state changed (launch/shutdown)
- Added automatic redirect if current mode is not local node
- Added opportunity to launch node if it is stopped
- Added error indicator if node launch failed. Error message contains the link to open brave://ipfs page for error diagnostic

Normal mode
![image](https://user-images.githubusercontent.com/2965009/116435899-86f73800-a854-11eb-910b-c14700f5b0c3.png)

Node is not launched
![image](https://user-images.githubusercontent.com/2965009/116549357-08ef6b80-a8fe-11eb-880f-856324fbf3fe.png)

Unable to start node
![image](https://user-images.githubusercontent.com/2965009/116549449-291f2a80-a8fe-11eb-9e06-ba56b0766f37.png)

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Go to brave://ipfs and use option to add/remove keys
- The item to setup IPNS keys should be visible only if current method is local node and the node is launched
- Try to start/stop node when the add/remove key dialog is opened, the dialog should be closed automatically and the button to start node should be shown
- Try to open brave://settings/ipfs/keys page when the node is not installed. 
      - If current method is local node you should be asked to start the node and see launch error. There's a linked text that can be clicked to navigate to new brave://ipfs tab 
      - If current method is not local node you must be redirected to brave://ipfs
- if you start node from keys page, UIs block to start node and to add keys should be switched automatically
